src/NLU.DevOps.ModelPerformance.Tests/TestCaseSource.cs

```diff
@@ -221,27 +227,40 @@ bool isEntityMatch(Entity expectedEntity, Entity actualEntity)
                             entity.EntityType);
                     }
 
-                    // "Semantic" entity value match test cases
-                    bool isEntityValueMatch(Entity actualEntity)
-                    {
-                        return entity.EntityValue == actualEntity.EntityValue;
-                    }
-
-                    if (entity.EntityValue != null && (actual == null || !actual.Any(isEntityValueMatch)))
-                    {
-                        yield return FalseNegative(
-                            $"FalseNegativeEntityValue('{entity.EntityType}', '{entity.EntityValue}', '{text}')",
-                            $"Actual utterance does not have entity value matching '{entity.EntityValue}'.",
-                            "Entity",
-                            entity.EntityType);
-                    }
-                    else if (entity.EntityValue != null)
+                    if (entity.EntityValue != null)
                     {
-                        yield return TruePositive(
-                             $"TruePositiveEntityValue('{entity.EntityType}', '{entity.EntityValue}', '{text}')",
-                             $"Both utterances have entity value '{entity.EntityValue}'.",
-                             "Entity",
-                             entity.EntityType);
+                        // "Semantic" entity value match test cases
+                        bool isEntityValueMatch(Entity actualEntity)
+                        {
+                            return entity.EntityValue == actualEntity.EntityValue;
+                        }
+
+                        var matchingEntitiesByValue = actual?.Where(isEntityValueMatch);
+                        var matchingValueCount = matchingEntitiesByValue?.Count() ?? 0;
+                        if (matchingValueCount == 0)
+                        {
+                            yield return FalseNegative(
+                                $"FalseNegativeEntityValue('{entity.EntityType}', '{entity.EntityValue}', '{text}')",
+                                $"Actual utterance does not have entity value matching '{entity.EntityValue}'.",
+                                "Entity",
+                                entity.EntityType);
+                        }
+                        else if (!matchingEntitiesByValue.Any(actualEntity => entity.EntityType == actualEntity.EntityType))
```

Comment:
We should just bump the entity type comparison into isEntityMatch, the logic below handles the false positives.